### PR TITLE
Allow loading model from in-memory byte-array

### PR DESCRIPTION
**Pull request by mika-fischer**

### CODE REVIEW

**Comment by mika-fischer** [comment]
I don't know if this is the preferred name for the new API function. If not, let me know and I'll change it.

**Comment by mika-fischer** [comment]
Anything I can do to move this along?

**Comment by snnn** [comment]
Please fix the compiler warning

**Comment by mika-fischer** [comment]
It should be fixed now, do I have to do anything to run the CI?

File: onnxruntime/core/session/inference_session.h [hunk 1]
```diff
@@ -179,6 +179,14 @@ class InferenceSession {
     */
   common::Status Load(std::istream& model_istream);
 
+  /**
+    * Load an ONNX model.
+    * @param model_data Model data buffer
+    * @param model_data_len Model data buffer size
+    * @return OK if success.
+    */
+  common::Status Load(const void* model_data, size_t model_data_len);
```
Comment (snnn) [inline onnxruntime/core/session/inference_session.h]: model_data_len should be in 'int' type ,not size_t. 
You're current implementation will silently discard data if it's larger than 2GB. It doesn't looks good.

Comment (mika-fischer) [reply onnxruntime/core/session/inference_session.h]: You're absolutely correct, that was stupid... It's fixed now.

**Comment by mika-fischer** [comment]
Anything I can do to move this forward?

**Comment by mika-fischer** [comment]
Pinging @snnn @RyanUnderhill @linkerzhang 

Anything I can do to move this forward?

**Comment by mika-fischer** [comment]
Another ping...

**Comment by snnn** [comment]
@linkerzhang @yufenglee Any comments on this PR?

**Comment by LauraRichter** [comment]
Is the ability to load a model from memory only available using the C++ api still? Or is it available with the python api now?

**Comment by mika-fischer** [comment]
@LauraRichter It seems in pyhton you can just pass bytes to the InferenceSession constructor, see https://github.com/microsoft/onnxruntime/blob/1d79926d273d01817ce93f001f36f417ab05f8a0/onnxruntime/python/session.py#L18

**Comment by LauraRichter** [comment]
Thank you for pointing that out! That's perfect

**Comment by dsplabs** [comment]
Is there a simple example demonstrating how to serialize the model to byte string?

**Comment by pranavsharma** [comment]
> Is there a simple python example demonstrating how to serialize the model to byte string?

ONNX model files are simply protobuf files, so you can use the python protobuf interface to serialize/deserialize, etc. Or you can follow the code in this file that relies on the onnx python pkg: https://github.com/onnx/onnx/blob/09a4e65bb098164491b021ffe563a559fbc1a808/onnx/__init__.py#L67.

**Comment by dsplabs** [comment]
Thank you @pranavsharma

**Pull request merged**